File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        result = []
        i = 0
        j = i + 1
        temp_count = len(temperatures)

        if temp_count < 2:
            return [0]
        while i < temp_count:
            if j >= temp_count:
                result.append(0)
                i = i+1
                j = i+1
            elif temperatures[j] > temperatures[i]:
                result.append(j - i)
                i = i + 1
                j = i + 1
            else:
                j = j + 1

        return result
        

File: Data Structures & Algorithms/eating-bananas/submission-1.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        long hr = (long) h;
        int left = 1;
        int right = Arrays.stream(piles).max().getAsInt();  
        int result = right;
        while (left<=right) {
            int mid = left + (right-left)/2;
            long hours = hours(piles, mid);
            if (hours <= hr) {
                result = mid;
                right = mid -1;
            } else left = mid+1;
        }
        return result;
    }

    private long hours(int[] arr, int rate) {
        long hours = 0;
        for (int i=0; i<arr.length; i++) hours += (arr[i]+rate-1)/rate;
        return hours;
    }
}
